src/TodosList.js

```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import { useParams } from "react-router-dom";
+import useTodo from "./hooks/useTodo";
+
+const TodosList = () => {
+  const { id } = useParams();
+  const { data, isLoading, error } = useTodo(id);
+
+  if (isLoading) {
+    return <div>로딩중</div>;
+  }
+
+  if (!isLoading && error) {
+    return <div>에러 발생</div>;
+  }
+
+  if (!isLoading && !error && data) {
+    return <div>{data.title}</div>;
+  }
+
+  return <>기다려주세요</>;
+};
```

Comment:
1. isLoading 이면 return 이라고 한 순간부터 `isLoading === false` 일 수밖에 없으므로, 이후 if 문에서 !isLoading 의 경우를 체크할 필요가 없어요. 아래처럼 수정하면 된다는 얘기입니다. error 의 경우도 마찬가지.

```suggestion
  if (isLoading) {
    return <div>로딩중</div>;
  }

  if (error) {
    return <div>에러 발생</div>;
  }

  if (data) {
    return <div>{data.title}</div>;
  }

  return <>기다려주세요</>;
};
```

2. 정상적인 return (데이터가 있는 경우) 를 가장 마지막에 두는 것이 일반적이에요. 그래야 맨 밑만 읽었을 때 정상 케이스를 확인할 수 있으니까요!

Reply:
왜 이 생각을 못했을가요,,,ㅎ 수정 완완입니당!! 감사합니다 🤧